common/k8s/src/middleware/metadata.rs

```diff
@@ -103,7 +104,32 @@ impl K8sMetadata {
             ListParams::default()
         };
 
-        let watched = StreamBackoff::new(watcher(api, params), ExponentialBackoff::default());
+        let watcher = watcher(api, params).map_ok(|ev| {
+            ev.modify(|pod| {
+                pod.managed_fields_mut().clear();
+                pod.status = None;
+
+                if let Some(pod_spec) = pod.spec.as_mut() {
+                    let orig = std::mem::take(pod_spec);
+                    let filtered_pod_spec = PodSpec {
+                        containers:
+                            orig
+                                .containers
+                                .into_iter()
+                                .map(|container| Container {
+                                    image: container.image,
+                                    ..Default::default()
+                                })
+                                .collect()
+                        ,
+                        ..Default::default()
+                    };
+                    *pod_spec = filtered_pod_spec;
+                }
+            })
+        });
+
+        let watched = StreamBackoff::new(watcher, ExponentialBackoff::default());
```

Comment:
@jakedipity 

This StreamBackoff may be our suspect in the missing k8s meta issue. The stream processing has silent errors (no logging).

> After any Err is emitted, the stream is paused for Backoff::next_backoff. The Backoff is reset on any Ok value. If Backoff::next_backoff returns None then the backing stream is given up on, and closed.